src/main/java/ru/ayaz/Chat.java

```diff
@@ -32,53 +28,21 @@ void start() {
         try {
             while (true) {
                 Socket socket = serverSocket.accept();
-                UserSocketHandler userSocketHandler = new UserSocketHandler(socket, messageDistributor);
-                User user = registerUser(userSocketHandler.getWriter(), userSocketHandler.getReader());
-                userSocketHandler.setNickname(user.getNickname());
-                messageDistributor.registerAtMessageDistributor(user, userSocketHandler);
+                UserSocketHandler userSocketHandler = new UserSocketHandler(socket, room);
+                room.registerAtChatRoom(userSocketHandler.getUser(), userSocketHandler);
                 new Thread(userSocketHandler).start();
             }
         } catch (IOException e) {
             System.out.println("User lost connection");
-        } catch (RegistrationFailException e) {
+        } catch (InvalidNicknameException e) {
             System.out.println(e.toString());
         }
     }
 
 
     private void startUserMessageDistributor() {
```

Comment:
&#x1F34E; **Issue**
I would pull the logic of room initialization from this method into constructor and rename this method, e.g.:
startUserMessageDistributor -> startServingVisitors